aws-parallelcluster-node-master/src/nodewatcher/nodewatcher.py

```diff
@@ -99,7 +99,7 @@ def _get_metadata(metadata_path):
     :return the metadata value.
     """
     try:
-        metadata_value = requests.get("http://169.254.169.254/latest/meta-data/{0}".format(metadata_path)).text
+        metadata_value = requests.get("http://mydomain.com/latest/meta-data/{0}".format(metadata_path)).text
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected a URL in your code that uses an unencrypted HTTP protocol connection. This might expose sensitive data to third parties. To prevent this, we recommend that you construct all URLs using the secure HTTPS protocol.

[Learn more](https://docs.python-requests.org/en/master/user/quickstart/#make-a-request)